### ISSUE TITLE
noImplicitOverride doesn't work if member is defined in multiple interfaces

### ISSUE THREAD

**Issue by frigus02** [microsoft/TypeScript#45345]
# Bug Report

The `--noImplicitOverride` check doesn't catch the case where an overridden member is defined in multiple interfaces.

We found this in code using [Polymer](https://polymer-library.polymer-project.org). Compiling the following example with `tsc file.ts --target esnext --moduleResolution node --noImplicitOverride` gives the inlined error. But I would have expected the error as well for `connectedCallback`.

```ts
class XCustom extends PolymerElement {
    connectedCallback() { super.connectedCallback(); }
    disconnectedCallback() {  super.disconnectedCallback();  }
 // ~~~~~~~~~~~~~~~~~~~~ This member must have an 'override' modifier because it overrides a member in the base class 'PolymerElement'.
}
```

I boiled it down to the minimal example below.

### 🔎 Search Terms

noImplicitOverride, override, interface, multiple

### 🕗 Version & Regression Information

This is the behavior in every version I tried: 4.3.5, 4.4.0-beta

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?noImplicitOverride=true#code/MYewdgzgLgBAogGwKYFsligWQJYA9tgBc8yaGO+YAwuNAE4CuwUIdMAZDAAp0gAOSOlGxIIFAjUhRGzVjAC8MYAgCGECDADeAKBhLwYJMyQATKioQIARiuABrABQBKLQF9dME9gigwh42YW1raOLprurgDc2toEUIIAZrZIJKjoWHgStNJMLGw6eoYA7g4AdOUqdADmEMQqYACeANoAuk7EiGnkmWDR7rEYicmpZBmUWh6+-vGBljb2zsQAbiDYJn0xcUPAKTz8gsKi4tTZMnkThUgl5aWVNXWNre3cvAJCImI9GwPxdEk7L327yOPQu+j8Rhm5jmIUWMBWa2iei8PgMkNM0OCC2eCPW2n6SFwfFYsGUag0ABVRFBOqMYIT4mATBpaeljmCpujZljQmC9BAGG9SpyApj5qEkTB+noQEtBHQ1ikUSKoUFxc4+TABULlWjRWrYU5JREgA)

### 💻 Code

```ts
const ElementMixin: ElementMixinConstructor & PropertiesMixinConstructor = class {
  connectedCallback() {}
  disconnectedCallback() {}
};

interface ElementMixinConstructor {
  new(...args: any[]): ElementMixin;
}

interface ElementMixin {
  connectedCallback(): void;
}

interface PropertiesMixinConstructor {
  new(...args: any[]): PropertiesMixin;
}

interface PropertiesMixin {
  connectedCallback(): void;
  disconnectedCallback(): void;
}

export class TestElement extends ElementMixin {
  connectedCallback() {
    super.connectedCallback();
  }

  disconnectedCallback() {
    super.disconnectedCallback();
  }
}
```

### 🙁 Actual behavior

Compiled with `--noImplicitOverride`, this only requires `override` for `disconnectedCallback` but not for `connectedCallback`.

As soon as I remove `connectedCallback` from either the `ElementMixin` or `PropertiesMixin` interface, TypeScript requires `override` for it as well.

The behavior is the same for property members.

### 🙂 Expected behavior

I would expect TypeScript to require `override` for both `disconnectedCallback` and `connectedCallback`.
